vite-project-kabinet/src/Content/PaginatedList.jsx

```diff
@@ -14,13 +16,53 @@ export const PaginatedList = ({items, itemsPerPage}) => {
         }
     };
 
+    // Fetch funkce
+    const getPostsWithCategoryName = async () => {
+        const reqPosts = await fetch(`/assets/constants/posts.json`)
+        const posts = await reqPosts.json()
+
+        const reqCategories = await fetch(`/assets/constants/categories.json`)
+        const categories = await reqCategories.json()
+
+        //chcem získať pole objektov, kde bude každý objekt obsahovať aj konkrétny názov kategórie
+        
+        const updatedPosts = posts.map((post) => {
+            const postCategsArray = post.categories
+            const newPostCategsArray = postCategsArray.map((catID) => {
+                const categObject = categories.find((category) => category.id === catID )
+                return categObject ? categObject.name : "kategória sa nenašla"
+            })
+
+            // Aby program správne fungoval, každý článok musí obsahovať vždy práve jednu kategóriu.
+            // Kabinet má aktuálne pár článkov takých, kde sú pridelené 2 kategórie
+
+            // todo: Dá sa nastaviť WP tak, aby uživateľ mohol zakliknúť vždy len jednu kategóriu ?
+            // todo: pridať žánre vo WP
+            // todo: nastaviť WP tak, aby šlo pridávať fotky
+            // todo: nastaviť WP tak, aby mal autor svoju vlastnú kolonku
+
+            return {
+                ...post,
+                categoryName: newPostCategsArray.join()
+            }
+        })
+
+        return updatedPosts
+    }
+
+    useEffect(() => {
+        getPostsWithCategoryName().then((data) => {
+            setItems(data)
+        })
+    }, [])
+
     return (
         <>
             <section className="card__container">
                 {items
                     .slice((currentPage - 1) * itemsPerPage, currentPage * itemsPerPage)
                     .map((item) => (
-                    <CardPreview key={item.id} id={item.id} genre={item.genre} title={item.title} author={item.author} content={item.content} image={item.image}/>
+                    <CardPreview key={item.id} id={item.id} genre={item.categoryName} title={item.title.rendered} author={item.author} content={item.excerpt.rendered} image={item.image}/>
```

Comment:
To že WordPress vrací strukturu `title.rendered` neznamená, že Tvoje codebase to tak musí taky mít. Pokud Ti to dává smysl, tak ok. Ale můžeš mít i ploší strukturu kde článek má jen `title`